framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -869,7 +869,11 @@ public AnnotationFormatter getAnnotationFormatter() {
      *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
-        return supportedQuals;
+        if (this.supportedQuals.isEmpty()) {
+            supportedQuals.addAll(createSupportedTypeQualifiers());
+            checkSupportedQuals();
+        }
+        return Collections.unmodifiableSet(supportedQuals);
```

Comment:
Yes, I meant the latter. The `create` methods return fresh objects, so we don't need to worry about modifying existing state. The `get` method returns the field, so we need to return only the unmodifiable view.